flake.nix

```diff
@@ -0,0 +1,34 @@
+{
+  inputs = {
+    utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, utils, ... }:
+    utils.lib.eachDefaultSystem (
+      system: let
+        pkgs = nixpkgs.legacyPackages.${system};
+      in {
+
+        packages.default = pkgs.stdenv.mkDerivation {
+          name = "chez-exe";
+          version = "0.0.1";
+          src = ./.;
+
+          buildInputs = with pkgs; [
+            chez
+            libuuid
+          ];
+
+          buildPhase = ''
+            mkdir -p $out/{bin,lib}
+            scheme --script gen-config.ss \
+            --prefix $out \
+            --bindir $out/bin \
+            --libdir $out/lib \
+            --bootpath ${pkgs.chez}/lib/csv9.5.8/ta6le \
```

Comment:
Is there a way to have the path determined by the version of chez? This will break when the chez version changes

Reply:
There is also a version attribute for each pkg in nixpkgs, you could inspect it like this:
```
nix repl '<nixpkgs>'
chez.version
```

Reply:
If you need more or less dependencies the buildInputs in the flake or the general inputs can be modified.
Also if you need to pin a specific dependency to a certain version, you could always host that dependency somewhere and fetch it via the flake inputs.<dependency>.url and pass it to the buildInputs.

The flake.lock file contains some metadata about the fetched dependency so if you get stuck on fetching old versions(e.g. by providing http endpoint which gets overwritten by multiple versions) 'nix flake update' will help.

If the build process / buildphase doesn't change, there isn't anything to do.